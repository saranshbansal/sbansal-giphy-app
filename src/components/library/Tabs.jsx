import PropTypes from "prop-types";
import React, { Component } from "react";
import Tab from "./Tab";

class Tabs extends Component {
  static propTypes = {
    children: PropTypes.instanceOf(Array).isRequired
  };

  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  state = {
    activeTab: this.props.children[0].props.label
  };

  handleClick(tab) {
    this.setState({ activeTab: tab });
  }

  render() {
    const {
      props: { children },
      state: { activeTab }
    } = this;

    return (
      <div className="tabs">
        <ol className="tab-list">
          {children.map(child => {
            const { label } = child.props;

            return (
              <Tab
                activeTab={activeTab}
                key={label}
                label={label}
                onClick={this.handleClick}
              />
            );
          })}
        </ol>
        <div className="tab-content">
          {children.map(child => {
            if (child.props.label !== activeTab) return undefined;
            return child.props.children;
          })}
        </div>
      </div>
    );
  }
}

export default Tabs;
