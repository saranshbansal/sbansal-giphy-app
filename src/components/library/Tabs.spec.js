import { mount, shallow } from "enzyme";
import React from "react";
import Tabs from "./Tabs";

describe("Tabs", () => {
  const dummyTabs = [
    <div label="label 1">
      {'label: label1'}
    </div>,
    <div label="label 2">
      {'label: label2'}
    </div>
  ];
  
  it("should render correctly.", () => {
    const wrapper = shallow(<Tabs children={dummyTabs} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render tabs with given data.", () => {
    const wrapper = mount(<Tabs children={dummyTabs} />);

    expect(wrapper.find('li')).toHaveLength(2);
  });
});
