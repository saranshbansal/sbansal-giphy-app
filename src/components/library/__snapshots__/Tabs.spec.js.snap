// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly. 1`] = `
<div
  className="tabs"
>
  <ol
    className="tab-list"
  >
    <Tab
      activeTab="label 1"
      key="label 1"
      label="label 1"
      onClick={[Function]}
    />
    <Tab
      activeTab="label 1"
      key="label 2"
      label="label 2"
      onClick={[Function]}
    />
  </ol>
  <div
    className="tab-content"
  >
    label: label1
  </div>
</div>
`;
